sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkAsyncHttpClient.java

```diff
@@ -89,7 +75,7 @@
             if (headers != null) {
                 for (HttpHeader header : headers) {
                     final String headerName = header.getName();
-                    if (!JDK12_RESTRICTED_HEADERS.contains(headerName)) {
+                    if (!restrictedHeaders.contains(headerName)) {
                         final String headerValue = header.getValue();
                         builder.setHeader(headerName, headerValue);
                     } else {
```

Comment:
There are several ways to have allowlist. Could you update the message here? Also, change `whitelist` wording.
Do we want to change the log level to WARNING?